pkgs/os-specific/linux/sysdig/default.nix

```diff
@@ -107,39 +114,37 @@ stdenv.mkDerivation rec {
     export KERNELDIR="${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
   '';
 
-  postInstall =
-    ''
-      # Fix the bash completion location
-      installShellCompletion --bash $out/etc/bash_completion.d/sysdig
-      rm $out/etc/bash_completion.d/sysdig
-      rmdir $out/etc/bash_completion.d
-      rmdir $out/etc
-    ''
-    + lib.optionalString (kernel != null) ''
-      make install_driver
-      kernel_dev=${kernel.dev}
-      kernel_dev=''${kernel_dev#${builtins.storeDir}/}
-      kernel_dev=''${kernel_dev%%-linux*dev*}
-      if test -f "$out/lib/modules/${kernel.modDirVersion}/extra/scap.ko"; then
-          sed -i "s#$kernel_dev#................................#g" $out/lib/modules/${kernel.modDirVersion}/extra/scap.ko
-      else
-          for i in $out/lib/modules/${kernel.modDirVersion}/{extra,updates}/scap.ko.xz; do
-            if test -f "$i"; then
-              xz -d $i
-              sed -i "s#$kernel_dev#................................#g" ''${i%.xz}
-              xz -9 ''${i%.xz}
-            fi
-          done
-      fi
-    '';
-
+  postInstall = ''
+    # Fix the bash completion location
+    installShellCompletion --bash $out/etc/bash_completion.d/sysdig
+    rm $out/etc/bash_completion.d/sysdig
+    rmdir $out/etc/bash_completion.d
+    rmdir $out/etc
+  '' + lib.optionalString (kernel != null) ''
+    make install_driver
+    kernel_dev=${kernel.dev}
+    kernel_dev=''${kernel_dev#${builtins.storeDir}/}
+    kernel_dev=''${kernel_dev%%-linux*dev*}
+    if test -f "$out/lib/modules/${kernel.modDirVersion}/extra/scap.ko"; then
+        sed -i "s#$kernel_dev#................................#g" $out/lib/modules/${kernel.modDirVersion}/extra/scap.ko
+    else
+        for i in $out/lib/modules/${kernel.modDirVersion}/{extra,updates}/scap.ko.xz; do
+          if test -f "$i"; then
+            xz -d $i
+            sed -i "s#$kernel_dev#................................#g" ''${i%.xz}
+            xz -9 ''${i%.xz}
+          fi
+        done
+    fi
+  '';
 
   meta = with lib; {
-    description = "A tracepoint-based system tracing tool for Linux (with clients for other OSes)";
+    description =
+      "A tracepoint-based system tracing tool for Linux (with clients for other OSes)";
     license = with licenses; [ asl20 gpl2 mit ];
-    maintainers = [maintainers.raskin];
-    platforms = ["x86_64-linux"] ++ platforms.darwin;
-    broken = kernel != null && versionOlder kernel.version "4.14";
+    maintainers = [ maintainers.raskin ];
+    platforms = [ "x86_64-linux" ] ++ platforms.darwin;
+    broken = kernel != null && ((versionOlder kernel.version "4.14") || (kernel.isHardened && lib.lists.findSingle (x: x == kernel.version) false true [ "4.19" "5.4" "lqx" ]));
```

Comment:
Is that working like I expect it/how can I ensure it the build output seems not to change (nixpkgs-review) if I mark it as broken?

Reply:
Sorry, I'm not familiar with marking packages broken or how it interacts with nixpkgs-review.

nit, feel free to ignore: IMO, this does seem involved enough that I'd appreciate a comment (or a link to an issue) describing the way(s) this build is broken (and ideally issues filed in the relevant upstream place(s) if there's any hope of fixing this).

Reply:
Marking them as broken comes from the last PR https://github.com/NixOS/nixpkgs/pull/293688#issuecomment-1987811365

It appears to me e.g. in case of 4.19 (only different options but same error for all 3 hardened-kernels) that some functions got stripped out (to harden) this causes 

`/nix/store/pi6cvclrd4ca1g6z525fsa71f4ayj8n7-linux-config-4.19.309.drv`

to fail with e.g.
```
warning: unused option: CFI_CLANG
warning: unused option: CONFIG_LDISC_AUTOLOAD
error: unused option: GCC_PLUGINS
error: unused option: GCC_PLUGIN_LATENT_ENTROPY
error: unused option: GCC_PLUGIN_RANDSTRUCT
error: unused option: GCC_PLUGIN_RANDSTRUCT_PERFORMANCE
warning: unused option: GCC_PLUGIN_STRUCTLEAK
warning: unused option: GCC_PLUGIN_STRUCTLEAK_BYREF_ALL
warning: unused option: INET_DIAG_DESTROY
warning: unused option: INET_MPTCP_DIAG
warning: unused option: INET_RAW_DIAG
warning: unused option: INET_TCP_DIAG
warning: unused option: INET_UDP_DIAG
warning: unused option: IOMMU_DEFAULT_DMA_LAZY
warning: unused option: IOMMU_DEFAULT_DMA_STRICT
warning: unused option: IO_STRICT_DEVMEM
warning: unused option: SECURITY_WRITABLE_HOOKS
warning: unused option: SLS
warning: unused option: STRICT_DEVMEM
warning: unused option: UBSAN_LOCAL_BOUNDS
```

SO I think sysdig is just not supported on these hardend-kernels due to their hardening. (assumption)